### ISSUE TITLE
`remove_all` does not remove cache

### ISSUE THREAD

**Issue by rkucmierowski** [mapproxy/mapproxy#927]
## Context
Is everything ok with `remove_all` cleanuping option for `file` cache?
According docs: _When set to true (...) it will remove complete level directories for file caches instead of comparing each tile with a timestamp_.
## Actual Behavior
But if `remove_all` is set to `True`,
`max_mtime = self.expire_timestamp(tile)` equals `0`  and tiles timestamp comparison is still performed,
so the condition `stale = int(tile.timestamp) <= max_mtime` evaluates to `False` and `is_cached()` returns `True`. 
https://github.com/mapproxy/mapproxy/blob/f3b976d04755955582f88b4e5c98bf3915632b4e/mapproxy/cache/tile.py#L228-L245
Then `is_stale()` returns `False`, which means that the tile has not expired and shouldn't be remove
https://github.com/mapproxy/mapproxy/blob/f3b976d04755955582f88b4e5c98bf3915632b4e/mapproxy/cache/tile.py#L247-L259

## Your Environment
    Version used: MapProxy 1.16
    Environment name and version: Python 3.7
    Server type and version:
    Operating System and version: Ubuntu

**Comment by simonseyock** [comment]
I assume you mean the `remove_all` option of the seeding cleanup tasks https://mapproxy.github.io/mapproxy/latest/seed.html#remove-all

There exists a test that covers exactly this case for the tile cache:
https://github.com/mapproxy/mapproxy/blob/master/mapproxy/test/system/test_seed.py#L193

The functions `is_cached` and `is_stale` are never called in this case. Can you provide further information about your setup and configuration?

**Comment by rkucmierowski** [comment]
Yes, I mean the `remove_all` option of the seeding cleanup tasks.

I realized that the problem is related to the `coverages` option. When a cleanup task has this specified, the `tilewalker_cleanup` is used and then the functions `is_cached` and `is_stale` are called.

It would be useful to have a test to check the `remove_all` option in combination with `coverages`.
